cubeb-core/src/ffi.rs

```diff
@@ -0,0 +1,539 @@
+// Copyright © 2017 Mozilla Foundation
```

Comment:
This looks like a combination of a bindgen generated file and hand editing.  I'm concerned that this will be difficult to maintain in the future, so I'd rather we simply use bindgen directly for all FFI level bindings (and let them deal with the bugs), then write cleaner Rust-y APIs on top of whatever bindgen spits out.

We don't need to block merging this PR on addressing this, but we do need to address it in the future.